ui/src/dashboards/components/DashboardEmpty.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {Component} from 'react'
+import {Cell} from 'src/types/dashboard'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+import uuid from 'uuid'
+
+import {addDashboardCellAsync} from 'src/dashboards/actions'
+import {GRAPH_TYPES} from 'src/dashboards/graphics/graph'
+
+interface Dashboard {
+  id: string
+  cells: Cell[]
+}
+
+interface Props {
+  dashboard: Dashboard
+  addDashboardCell: (dashboard: Dashboard, cell?: Cell) => void
+}
+
+const mapDispatchToProps = dispatch => ({
+  addDashboardCell: bindActionCreators(addDashboardCellAsync, dispatch),
+})
+
+@connect(null, mapDispatchToProps)
+class DashboardEmpty extends Component<Props> {
+  constructor(props) {
+    super(props)
+  }
+
+  public handleAddCell = type => () => {
```

Comment:
does any component outside of this one calle handleAddCell? consider making this private instead of public

Reply:
Yeah the dashboard header has an `Add Cell` button that calls the action as well